astropy/cosmology/io/mapping.py

```diff
@@ -81,6 +86,9 @@ class itself.
         FlatLambdaCDM(name="Planck18", H0=67.7 km / (Mpc s), Om0=0.31,
                       Tcmb0=0 K, Neff=3.05, m_nu=None, Ob0=0.049)
     """
+    if not isinstance(map, Mapping):
```

Comment:
How about instead use
```
params = dict(map)  # so we are guaranteed to have a poppable map
```
That then checks the input and makes the copy as well. Plus, no distractions.